### ISSUE TITLE
group_rows() does not work with non-unique rows

### ISSUE THREAD

**Issue by rabutler** [haozhu233/kableExtra#103]
Thanks for this awesome package!

I think I've found an issue with `group_rows()` with latext output (not sure about html as I haven't tested it). 

If there are rows that are not unique, `group_rows()` does not properly indent them. 

Ex:
```
library(knitr)
library(kableExtra)
library(dplyr)

t2 <- data.frame(a = 1:7, b = month.abb[1:7])

# this works as expected:
kable(t2, format = "latex", booktabs = TRUE) %>% 
  group_rows("G1", 1, 2) %>% 
  group_rows("G2", 3,7)
```
But if you have another January row in the second group, it does not indent it:
```
t2 <- add_row(t2, a = 1, b = "Jan")

# doesn't work:
kable(t2, format = "latex", booktabs = TRUE) %>% 
  group_rows("G1", 1, 2) %>% 
  group_rows("G2", 3, 8)
```

The output is: 

> \begin{tabular}{rl}
\toprule
a & b\\
\midrule
\addlinespace[0.3em]
\multicolumn{2}{l}{\textbf{G1}}\\
\hspace{1em}\hspace{1em}1 & Jan\\
\hspace{1em}2 & Feb\\
\addlinespace[0.3em]
\multicolumn{2}{l}{\textbf{G2}}\\
\hspace{1em}3 & Mar\\
\hspace{1em}4 & Apr\\
\hspace{1em}5 & May\\
\hspace{1em}6 & Jun\\
\hspace{1em}7 & Jul\\
1 & Jan\\
\bottomrule
\end{tabular}

And I would expext the last "1 & Jan\\" row to also be indented. 

It's worth noting that it also does not work if there are non-unique rows in the same group, although it's a little harder to imagine when this might need to occur. The first case certainly seems likely though as you might have similar results for two different groups.

```
t2 <- data.frame(a = c(1, 1:4), b = month.abb[c(1, 1:4)])

# doesn't seem to work, the second Jan row is not indented 
kable(t2, format = "latex", booktabs = TRUE) %>% 
  group_rows("G1", 1, 3) 
```
Thanks!

**Comment by haozhu233** [comment]
Hi, @rabutler, I think you probably are using an old version of `kableExtra`. I fixed this bug in ver 0.6.0. :)

If you update you package, this bug should be gone. 

Hao

**Comment by rabutler** [comment]
Thanks @haozhu233. You are correct. My apologies for not ensuring I was using the latest version. Thanks again!
